### ISSUE TITLE
not working on the react@18.2.0 version 

### ISSUE THREAD

**Issue by vilashpixelnx** [ethanselzer/react-image-magnify#170]
not working on the react@18.2.0 version

**Comment by vilashpixelnx** [comment]
$ npm install react-image-magnify
npm ERR! code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR!
npm ERR! While resolving: frontend@0.1.0
npm ERR! Found: react@18.2.0
npm ERR! node_modules/react
npm ERR!   react@"^18" from the root project
npm ERR!
npm ERR! Could not resolve dependency:
npm ERR! peer react@"~0.14.9 || ^15.0.0 || ^16.0.0" from react-image-magnify@2.7.4
npm ERR! node_modules/react-image-magnify
npm ERR!   react-image-magnify@"*" from the root project
npm ERR!
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.

**Comment by CKsabari2001** [comment]
send me the link for the source code

**Comment by vilashpixelnx** [comment]
https://www.npmjs.com/package/react-image-zoom

**Comment by vilashpixelnx** [comment]
> send me the link for the source code

https://www.npmjs.com/package/react-image-zoom

**Comment by Vivek-04022001** [comment]
> send me the link for the source code

Why do you need the source code? Don't you see this is the compatibility issue. Are you the repo maintainer?

**Comment by Vivek-04022001** [comment]
> npm ERR! this command with --force or --legacy-peer-deps

Use --force or --legacy-peer-deps to run this package.

**Comment by mohamed-ahmed-yousef** [comment]
> $ npm install react-image-magnify npm ERR! code ERESOLVE npm ERR! ERESOLVE unable to resolve dependency tree npm ERR! npm ERR! While resolving: frontend@0.1.0 npm ERR! Found: react@18.2.0 npm ERR! node_modules/react npm ERR! react@"^18" from the root project npm ERR! npm ERR! Could not resolve dependency: npm ERR! peer react@"~0.14.9 || ^15.0.0 || ^16.0.0" from react-image-magnify@2.7.4 npm ERR! node_modules/react-image-magnify npm ERR! react-image-magnify@"*" from the root project npm ERR! npm ERR! Fix the upstream dependency conflict, or retry npm ERR! this command with --force or --legacy-peer-deps npm ERR! to accept an incorrect (and potentially broken) dependency resolution.


The same issues happen to me. Did you find a solution?

**Comment by john-subba** [comment]
You can check this out also. Forked the original repo and converted everything to functional component. Should probably work like the previous original one.

https://www.npmjs.com/package/react-image-magnifier-waft 

And I want to thank the main author of the package for awesome package. Good Job.

**Comment by ChetSocio** [comment]
Hello Guys, BatchNepal is creating a similar Open Source package named: Easy magnify,.
Package Link: [https://www.npmjs.com/package/easy-magnify](https://www.npmjs.com/package/easy-magnify)
Github Link : https://github.com/ChetSocio/easy-magnify

Within few days it will have some breaking changes to make it compatible with react 18 and typescript support with functional components. Keep your eye on it and dont forget to star it.

**Comment by john-subba** [comment]
> Hello Guys, BatchNepal is creating a similar Open Source package named: Easy magnify,. Package Link: https://www.npmjs.com/package/easy-magnify Github Link : https://github.com/ChetSocio/easy-magnify
> 
> Within few days it will have some breaking changes to make it compatible with react 18 and typescript support with functional components. Keep your eye on it and dont forget to star it.

I had previously used your package but class based code gave some error in my project. So i forked and kind of maintained it. Looking forward to your release and maybe will switch to your package

**Comment by ChetSocio** [comment]
Yes, there was no plan to update it but after noticing it getting regular around 1k weekly downloads, I kinda felt i should maintain it. I am adding features and typescript with react 18 support. Updated package will be published in around 2-3 days.  Note: there are breaking changes

**Comment by john-subba** [comment]
> Yes, there was no plan to update it but after noticing it getting regular around 1k weekly downloads, I kinda felt i should maintain it. I am adding features and typescript with react 18 support. Updated package will be published in around 2-3 days. Note: there are breaking changes

Yeah i busted my ass to transfer it to functional component myself because i needed that magnify feature in my product description in multiple ecommerce projects. Since it had defaultProps issues had to change it myself. I will look into your changes and switch accordingly. Thanks for your effort. I always respect guys who contribute to open-source projects, makes dev lives whole lot easier.

**Comment by abhishekkumbhani** [comment]
Awaiting for latest updates. :)

**Comment by ChetSocio** [comment]
Yo guys! My package has been upgraded and ready for production use. Use its latest version. Its expandable, and reusable with decent performance.

Docs and real-time demo will be at : [https://easy-magnify.batchnepal.com](https://easy-magnify.batchnepal.com)
